### PR TITLE
Vscode extension

### PR DESCRIPTION
**Pull request by GJMcGowan**
Fixes #6131

WIP on vscode extension for sqlx files

### CODE REVIEW

File: vscode/sql.grammar.json [hunk 1]
```diff
@@ -0,0 +1,475 @@
+[
```
Comment (lewish) [inline vscode/sql.grammar.json:1]: Pretty sure you can delete this file, and sql.plist, we don't actually use them

Comment (lewish) [reply vscode/sql.grammar.json:1]: The SQLX grammar references the built in langauges directly

File: vscode/server.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+connection.onRequest("compile", async () => {
+  const _ = compileAndValidate();
+});
+
+documents.onDidSave(change => {
+  const _ = compileAndValidate();
+});
+
+async function compileAndValidate() {
+  const compileResult = await getProcessResult(exec("dataform compile --json"));
```
Comment (lewish) [inline vscode/server.ts:47]: I don't quite understand how this works - you don't pass the project directory in, the PWD is probably not the one you want?

Comment (GJMcGowan) [reply vscode/server.ts:47]: I don't understand how it works either. But it does 🤷‍♂️

**Pull request merged**